gsl_projet/views.py

```diff
@@ -84,19 +87,58 @@ def get_context_data(self, **kwargs):
         return context
 
     def get_ordering(self):
-        sorting = self.request.GET.get("tri")
-        available_sortings = {
-            "date_de_depot": "dossier_ds__ds_date_depot",
-            "cout_total": "dossier_ds__finance_cout_total",
+        ordering_map = {
+            "date_desc": "-dossier_ds__ds_date_depot",
+            "date_asc": "dossier_ds__ds_date_depot",
+            "cout_desc": "-dossier_ds__finance_cout_total",
+            "cout_asc": "dossier_ds__finance_cout_total",
+            "commune_desc": "-address__commune__name",
+            "commune_asc": "address__commune__name",
         }
-        if sorting in available_sortings:
-            return available_sortings.get(sorting)
+
+        tri = self.request.GET.get("tri")
+        return ordering_map.get(tri, None)
 
     def get_queryset(self):
         qs = Projet.objects.for_user(self.request.user).filter(
             dossier_ds__ds_date_depot__gte=datetime.date(2024, 9, 1)
         )
+
+        # Filtre par dispositif
+        dispositif = self.request.GET.get("dispositif")
+        if dispositif:
+            qs = qs.filter(dossier_ds__demande_dispositif_sollicite=dispositif)
+
+        # Filtre par coût total minimum
+        cout_min = self.request.GET.get("cout_min")
+        if cout_min:
```

Comment:
Idée : vérifier que cout_min est bien un entier, vu que c'est une donnée utilisateur (donc on fait pas confiance).
```suggestion
        if cout_min and cout_min.isnumeric():
            cout_min = int(cout_min)
```